pkg/services/vimmachine.go

```diff
@@ -215,9 +223,14 @@ func (v *VimMachineService) GetHostInfo(ctx context.Context, machineCtx capvcont
 func (v *VimMachineService) findVSphereVM(ctx context.Context, vimMachineCtx *capvcontext.VIMMachineContext) (*infrav1.VSphereVM, error) {
 	// Get ready to find the associated VSphereVM resource.
 	vm := &infrav1.VSphereVM{}
+	name, err := generateVMObjectName(vimMachineCtx, vimMachineCtx.Machine.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	vmKey := types.NamespacedName{
 		Namespace: vimMachineCtx.VSphereMachine.Namespace,
-		Name:      generateVMObjectName(vimMachineCtx, vimMachineCtx.Machine.Name),
+		Name:      name,
```

Comment:
This get lookup will never work with a generated name containing `{{.random}}` because the name never matches now.

So for me it does not make sense to use the new `generateVMObjectName` here inside `findVSphereVM`.

Reply:
We may need to change findVSphereVM to find the VSphereVM by OnwerReferences instead?!

This e.g. breaks the deletion flow in some way.

Reply:
We are not supporting {{.random}} function .. we only support `trunc` and `trimSuffix`

Reply:
ah then it makes sense 😂 , sorry context switching to a similar topic which includes the random part.